src/Database/Schema/TableSchemaInterface.php

```diff
@@ -225,9 +225,9 @@ public function indexes(): array;
      * @param string $name The name of the constraint.
      * @param array|string $attrs The attributes for the constraint.
      *   If string it will be used as `type`.
-     * @return \Cake\Database\Schema\TableSchemaInterface
+     * @return self
      */
-    public function addConstraint(string $name, $attrs);
+    public function addConstraint(string $name, $attrs): self;
```

Comment:
Arent those chaining in a way as well? Those should also not have typehints here.

Reply:
Return typehint removed.